lua/nvim-tree/explorer/explore.lua

```diff
@@ -40,22 +40,24 @@ local function populate_children(handle, cwd, node, git_status, parent)
     ---@type uv.fs_stat.result|nil
     local stat = vim.loop.fs_stat(abs)
     local filter_reason = parent.filters:should_filter_as_reason(abs, stat, filter_status)
-    if filter_reason == FILTER_REASON.none and not nodes_by_path[abs] and Watcher.is_fs_event_capable(abs) then
-      local child = nil
-      if t == "directory" and vim.loop.fs_access(abs, "R") then
-        child = builders.folder(node, abs, name, stat)
-      elseif t == "file" then
-        child = builders.file(node, abs, name, stat)
-      elseif t == "link" then
-        local link = builders.link(node, abs, name, stat)
-        if link.link_to ~= nil then
-          child = link
+    if filter_reason == FILTER_REASON.none and not nodes_by_path[abs] then
+      if Watcher.is_fs_event_capable(abs) then
+        local child = nil
+        if t == "directory" and vim.loop.fs_access(abs, "R") then
+          child = builders.folder(node, abs, name, stat)
+        elseif t == "file" then
+          child = builders.file(node, abs, name, stat)
+        elseif t == "link" then
+          local link = builders.link(node, abs, name, stat)
+          if link.link_to ~= nil then
+            child = link
+          end
+        end
+        if child then
+          table.insert(node.nodes, child)
+          nodes_by_path[child.absolute_path] = true
+          explorer_node.update_git_status(child, node_ignored, git_status)
         end
-      end
-      if child then
-        table.insert(node.nodes, child)
-        nodes_by_path[child.absolute_path] = true
-        explorer_node.update_git_status(child, node_ignored, git_status)
       end
     else
```

Comment:
This was added recently and appears it will execute despite `is_fs_event_capable` fail.

I reckon we should wrap that test around the whole block, something like:
```lua
    local abs = utils.path_join { cwd, name }
    if Watcher.is_fs_event_capable(abs) then
```

Defering to @evertonse on this one :)

Reply:
We could live dangerously and optimise this a little, only testing when `t == "directory"` as per:
https://github.com/nvim-tree/nvim-tree.lua/blob/15942df2bb8aa4a3006e0b32ef805fee9fa172d5/lua/nvim-tree/explorer/reload.lua#L132